apps/src/sites/code.org/pages/public/teacher-dashboard/index.js

```diff
@@ -366,7 +376,15 @@ function main() {
           $.each(resultStudents, function (index, student) {
             $scope.section.students.unshift(student);
           });
-        }).$promise.catch($scope.genericError);
+        }).$promise.then(() => {
+          // If we went from zero to more than zero students, rerender login
+          // type controls so the correct options are available.
+          // TODO: Once everything is React this should become unnecessary.
+          if (newStudents.length === $scope.section.students.length) {
+            unmountLoginTypeControls();
+            renderLoginTypeControls($scope.section.id);
```

Comment:
This is lazy, but the easiest way to get updated section info into Redux so the paragraph can correctly reflect available options, is to just re-render the component, which triggers a new async load from the server.  I don't feel _too_ bad about this since all this angular is in theory on the way out... but it is kind of ugly.